src/runMocha.js

```diff
@@ -16,7 +16,10 @@ const runMocha = ({ config, testPath, globalConfig }, workerCallback) => {
 
       runner.on('test end', test => tests.push(test));
       runner.on('pass', test => passes.push(test));
-      runner.on('fail', test => failures.push(test));
+      runner.on('fail', (test, err) => {
+        test.err = err;
+        failures.push(test);
```

Comment:
I wonder if we should make them all `Set` instead of `Array`, and then simply call `tests.add(test)` here